sdv/_utils.py

```diff
@@ -348,3 +348,83 @@ def check_sdv_versions_and_warn(synthesizer):
 
             message = f'{message} {static_message}'
             warnings.warn(message, SDVVersionWarning)
+
+
+def _check_greater_version(current_version, synthesizer_version):
+    """Check if the synthesizer version is greater than the current version.
+
+    Args:
+        current_version (str):
+            The current version to compare against, formatted as a string with major, minor, and
+            revision parts separated by periods (e.g., "1.0.0").
+        synthesizer_version (str):
+            The synthesizer version to compare, formatted as a string with major, minor, and
+            revision parts separated by periods (e.g., "1.0.0")
+
+    Returns:
+        bool:
+            True if the synthesizer version is greater than the current version, False otherwise.
+    """
+    current_version = current_version.split('.')
+    synthesizer_version = synthesizer_version.split('.')
+    for current_v, synth_v in zip(current_version, synthesizer_version):
+        try:
+            current_v = int(current_v)
+            synth_v = int(synth_v)
+            if current_v > synth_v:
+                return False
+
+            if synth_v > current_v:
+                return True
+        except Exception:
+            pass
+
+    return False
+
+
+def check_synthesizer_version(synthesizer):
+    """Check if the current synthesizer version is greater than the package version.
+
+    Args:
+        synthesizer (BaseSingleTableSynthesizer or BaseMultiTableSynthesizer):
```

Comment:
Does it work for PAR also?

Reply:
Yes, I will correct the `BaseSingleTableSynthesizer` to `BaseSynthesizer` since that is the main class being inherited.